src/Plugin/Filter/TokenFilter.php

```diff
@@ -89,7 +104,23 @@ public static function create(ContainerInterface $container, array $configuratio
    * {@inheritdoc}
    */
   public function process($text, $langcode) {
-    return new FilterProcessResult($this->token->replace($text), [], ['langcode' => $langcode]);
+    $data = [];
+
+    // Attempt to figure out the current context based on the current backtrace.
```

Comment:
While it might be possible, it doesn't appear that it will be out of the box as for at least the Node context is tied directly to the route, which means that while it would work on a Node page, it wouldn't work when the filter is processing a node being rendered by a view, or similar.

However, this is only the result of 5-10 minutes looking into a system I have not previously used, so I could be gravely mistaken.